### DISCUSSION TITLE
Issues with MultiSelect Combobox

### DISCUSSION THREAD

**Discussion by Sneha-Uppalapati** [twilio-labs/paste#3448] [Q&A] [answered]
Hi team, 
Here are the 2 issues that we are seeing with MultiSelect Combobox:

- We see errors for test cases for combobox. 
![image](https://github.com/twilio-labs/paste/assets/110847862/5d0edc0a-7c00-45ac-9ccc-f8ce817f51fd)

- We are trying to use Twilio paste multiselect combobox, however in one of our use cases, the combo box doesn’t show up scroll bar, even though it should show it by default. Really appreciate your insights on this. Thank you.

<img width="1082" alt="image" src="https://github.com/twilio-labs/paste/assets/110847862/8ee31971-84bc-405e-887c-badc3564ade5">

**Comment by TheSisb** [comment]
Hi @Sneha-Uppalapati,

Nothing really stands out from the information provided. Can you provide us the full stack trace and reproduction?

**Reply by Sneha-Uppalapati** [reply]
We are unable to provide the scroll for the component, with a fixed footer being in the bottom. 

https://github.com/twilio-labs/paste/assets/110847862/bad9a343-c32c-4325-8914-9676e9fc0a97

**Reply by TheSisb** [reply]
Hi @Sneha-Uppalapati,

I'm not sure if this video and the original test issue are related, but for the video provided you need to decrease your fixed footer zIndex. We're using the [highest available zindex in the system (zIndex90)](https://paste.twilio.design/tokens/list#heading-z-indices) for the combobox dropdown menu.  I recommend changing your fixed footer to use zIndex10 or zIndex20.

**Reply by Sneha-Uppalapati** [reply]
Thank you for your response. Our main issue is the UI scroll, so wanted to prioritize that over testing. For the footer we currently use Evergreen components

```
import NavigationBar from 'evergreen-draft/navigation-bar/src/NavigationBar'
import { Button, minorScale } from 'evergreen-ui'

<NavigationBar
        position={'sticky'}
        bottom={0}
        backgroundColor="tint1"
        paddingX={minorScale(7)}
        rightView={
          <Button
            isLoading={upsertLoading}
            appearance="primary"
            onClick={handleClickSave}
            disabled={!isSaveButtonDisabled}
          >
            Save Twilio account
          </Button>
        }
      />
```

Setting zIndex for the `navigationBar` didn't fix the issue.

**Reply by SiTaggart** [reply]
Hi

I just want to clarify your wording to help explain what you are observing better to us.

Are you saying that when a Combobox is at the bottom of the document, at the end of the scrolling window, when it is open the list of options are cut off at the bottom of the window because the window won't scroll any further to see them. Essentially, the list of Combobox options doesn't cater to the position it is in, in regard to the Browser window.

Is that correct?

If so, that is a known feature limitation right now. The list of options does not position itself to the window, only to the combobox itself. So if the window isn't big enough the options list will get cut off.

**Reply by Sneha-Uppalapati** [reply]
Yes that is exactly what we were looking for. We were under an assumption that the dropdown should cater its position based on the browser window, but thank you for confirming that its a feature limitation for now. 
Please let us know if it is possible to do either of these, as we couldn't find them in the documentation:
1. Can we display the dropdown upwards rather than downwards? 
2. Can we provide the number of rows that we want in the dropdown, so that scroll can start working after 2 rows?

**Reply by SiTaggart** [reply]
Looks like we're going to just fix it https://github.com/twilio-labs/paste/pull/3459

**Reply by Sneha-Uppalapati** [reply]
Is there a timeline by when this will be merged, and what version of paste would that come in?

**Reply by SiTaggart** [reply]
We'll see how we go, maybe today and what ever the next semver version of Paste that goes out, likely `20.2.0`.

**Reply by TheSisb** [reply, marked as answer]
Now released in 20.2.0.
